### PR TITLE
update serviceaccount cmds to use externals

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @deads2k @soltysh

### CODE REVIEW

**Comment by juanvallejo** [comment]
cc @deads2k

File: pkg/serviceaccounts/util/helpers.go [hunk 1]
```diff
@@ -14,26 +14,56 @@ const (
...
-	return apihelpers.GetName(serviceAccount.Name, "token-", maxSecretPrefixNameLength)
+func GetTokenSecretNamePrefix(serviceAccountName string) string {
+	return apihelpers.GetName(serviceAccountName, "token-", maxSecretPrefixNameLength)
 }
 
-func GetDockercfgSecretNamePrefixV1(serviceAccount *v1.ServiceAccount) string {
-	return apihelpers.GetName(serviceAccount.Name, "dockercfg-", maxSecretPrefixNameLength)
+// IsValidServiceAccountToken returns true if the given secret contains a service account token valid for the given service account
+// TODO, these should collapse onto the upstream method
+func IsValidServiceAccountToken(serviceAccount *kapi.ServiceAccount, secret *kapi.Secret) bool {
```
Comment (deads2k) [inline pkg/serviceaccounts/util/helpers.go]: This just pushed your problem around.  It didn't fix it.  Leave this one where it lies and make your external one.

File: pkg/serviceaccounts/util/helpers.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package util
 
 import (
-	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	kapi "k8s.io/kubernetes/pkg/apis/core"
```
Comment (deads2k) [inline pkg/serviceaccounts/util/helpers.go]: this dependency needs to go.

**Comment by juanvallejo** [comment]
@deads2k thanks, review comments addressed

**Comment by deads2k** [comment]
And now I've looked at usage. Looks like disjoint sets of helpers. One set for `oc` (move to `oc/lib`) and one set for a single controller (move to point of use) and one also used by test/util (switch to upstream).

**Comment by juanvallejo** [comment]
@deads2k 
> And now I've looked at usage. Looks like disjoint sets of helpers. One set for oc (move to oc/lib) and one set for a single controller (move to point of use) and one also used by test/util (switch to upstream).

Thanks, created oc/lib/serviceaccount for use with oc commands, moved controller-specific helpers to point of use, and switched that test to use the upstream helper

**Comment by juanvallejo** [comment]
@soltysh friendly ping

**Pull request merged**